xComputer: Fixes #44

markgossa opened the pull request:
**Pull Request (PR) description**
Pull Request to fix issue 44 with xComputer.psm1 where the Get-ComputerDomain function uses a command which LocalSystem doesn't have rights to. Changed to use this instead:

`Get-WmiObject -Class Win32_ComputerSystem`

**This Pull Request (PR) fixes the following issues:**
Fixes #44

**Task list:**
- [x] Change details added to Unreleased section of CHANGELOG.md?
- [ ] Added/updated documentation, comment-based help and descriptions in .schema.mof files where appropriate?
- [ ] Examples appropriately updated?
- [x ] New/changed code adheres to [Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?
- [x] [Unit and (optional) Integration tests](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md) created/updated where possible?

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xcomputermanagement/113)
<!-- Reviewable:end -->

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/PowerShell/xComputerManagement?pullRequest=113) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/PowerShell/xComputerManagement?pullRequest=113) before we can accept your contribution.<br/><br/>:x: markgossa [sign now](https://cla.opensource.microsoft.com/PowerShell/xComputerManagement?pullRequest=113)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/PowerShell/xComputerManagement?pullRequest=113) it.</sub>

PlagueHO commented:
Really great job @markgossa - for a first PR this is really great work! :grin: 

Couple of minor tweaks, nothing major at all.

There is one hard part to do though :grin: this new code you've added is really good but it has now actually made this function testable! So we need to add some unit tests for it - and these are usually the trickiest bit.

The tests will allow us to validate that the code works as expected (a.k.a. functional testing) as well in future that it continues to work (a.k.a. regression testing). We use a Pester (a PS Module) to execute the tests. Here is a great article to help you get started: http://duffney.io/GettingStartedWithPester .

So what you'll need to do is add some unit tests to this file here:
https://github.com/PowerShell/xComputerManagement/blob/5a468b311115b9c6314e4576d4f0ad97ba0fe7a5/Tests/Unit/MSFT_xComputer.Tests.ps1#L928

You'd add a section something like this:
```powershell
Context "$($script:DSCResourceName)\Get-ComputerDomain" {
	It 'Should not throw when NetBios is not specified' {
		{
			Get-ComputerDomain -Verbose
		} | Should Not Throw
	}

	It 'Should ....' {
		# Tests
	}

	# More tests here...

	It 'Should not throw when NetBios is specified' {
		{
			Get-ComputerDomain -NetBios -Verbose
		} | Should Not Throw
	}	
}
```

This will then cause the unit test coverage to increase. I'm more than happy to help you with all this as I know Pester can have a bit of a steep learning curve (but well worth learning). I apologize if you already know all this :grin:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[README.md, line 216 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhmrqLAy509wcjHeh:-KyhhmrrHcvnNdWGx0-s:b-qzsni0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L216)):*
> ```Markdown
> * xOfflineDomainJoin:
>   * Updated to meet HQRM guidelines.
> ```

Can you please remove this blank line - it'll trip up the automated release note generation up :grin:

---

*[README.md, line 217 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhgTneI983-sUcM8I:-KyhhgToFxkkCzfB6_mg:bivf2nk) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L217)):*
> ```Markdown
>   * Updated to meet HQRM guidelines.
> 
> * xComputer
> ```

Can you pop a colon on the end of this line for consistency.

---

*[README.md, line 218 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhiZzPhSoBWbvILPW6:-KyhiZzPhSoBWbvILPW7:b-33f7m4) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L218)):*
> ```Markdown
> 
> * xComputer
>   * Resolved bug in Get-ComputerDomain where LocalSystem doesn't have rights to the domain
> ```

Can you reduce this line down to 80 chars - one of the tests is failing (https://ci.appveyor.com/project/PowerShell/xcomputermanagement/build/1.9.326.0#L385) because the README.MD lines need to be a maximum of 80 chars.

Just do something like this:
```markdown
  * Resolved bug in Get-ComputerDomain where LocalSystem doesn't have rights
  to the domain
```

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 568 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhicDOVPYVg7yPkqu5:-KyhicDOVPYVg7yPkqu6:b2v8blg) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L568)):*
> ```PowerShell
>         else
>         {
>             if((Get-WmiObject -Class Win32_ComputerSystem).PartOfDomain -eq $true)
> ```

WMI cmdlets is being phased out in favor of CIM cmdlets. So can you instead use:
```powershell
Get-CimInstance -ClassName Win32_ComputerSystem
```

What you could also do to improve performance is to assign the result of the `Get-CimInstance` to a variable and use that. E.g.
```powershell
$computerSystem = Get-CimInstance -ClassName Win32_ComputerSystem
if ($computerSystem.PartOfDomain -eq $true)
{
  $domainName = $computerSystem.Domain
}
else
{
  $domainName = $null
}
```

This will reduce the number of CIM lookups to 1 instead of 2. CIM (and WMI) lookups are quite slow.

Also, please add a space after if and before open bracket.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-KyhhCRQaSBVmIgLtZYE:bk39kn)*
<!-- Sent from Reviewable.io -->

markgossa commented:
Hi Daniel,

That's great thanks. I started learning Pester yesterday and will give it a go.

The other changes are quite straight forward so should be okay.

I'll send you another pull request when I have this all done.

Thanks.

Mark


-------- Original message --------
From: Daniel Scott-Raynsford <notifications@github.com>
Date: 12/11/2017 01:21 (GMT+00:00)
To: PowerShell/xComputerManagement <xComputerManagement@noreply.github.com>
Cc: Mark Gossa <mark@gossa.co.uk>, Mention <mention@noreply.github.com>
Subject: Re: [PowerShell/xComputerManagement] xComputer: Issue 44 (#113)


Really great job @markgossa<https://github.com/markgossa> - for a first PR this is really great work! 😁

Couple of minor tweaks, nothing major at all.

There is one hard part to do though 😁 this new code you've added is really good but it has now actually made this function testable! So we need to add some unit tests for it - and these are usually the trickiest bit.

The tests will allow us to validate that the code works as expected (a.k.a. functional testing) as well in future that it continues to work (a.k.a. regression testing). We use a Pester (a PS Module) to execute the tests. Here is a great article to help you get started: http://duffney.io/GettingStartedWithPester .

So what you'll need to do is add some unit tests to this file here:
https://github.com/PowerShell/xComputerManagement/blob/5a468b311115b9c6314e4576d4f0ad97ba0fe7a5/Tests/Unit/MSFT_xComputer.Tests.ps1#L928

You'd add a section something like this:

Context "$($script:DSCResourceName)\Get-ComputerDomain" {
        It 'Should not throw when NetBios is not specified' {
                {
                        Get-ComputerDomain -Verbose
                } | Should Not Throw
        }

        It 'Should ....' {
                # Tests
        }

        # More tests here...

        It 'Should not throw when NetBios is specified' {
                {
                        Get-ComputerDomain -NetBios -Verbose
                } | Should Not Throw
        }
}

This will then cause the unit test coverage to increase. I'm more than happy to help you with all this as I know Pester can have a bit of a steep learning curve (but well worth learning). I apologize if you already know all this 😁

________________________________

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

________________________________

README.md, line 216 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhmrqLAy509wcjHeh:-KyhhmrrHcvnNdWGx0-s:b-qzsni0> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L216>):

* xOfflineDomainJoin:
  * Updated to meet HQRM guidelines.

Can you please remove this blank line - it'll trip up the automated release note generation up 😁

________________________________

README.md, line 217 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhgTneI983-sUcM8I:-KyhhgToFxkkCzfB6_mg:bivf2nk> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L217>):

  * Updated to meet HQRM guidelines.

* xComputer

Can you pop a colon on the end of this line for consistency.

________________________________

README.md, line 218 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhiZzPhSoBWbvILPW6:-KyhiZzPhSoBWbvILPW7:b-33f7m4> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L218>):

* xComputer
  * Resolved bug in Get-ComputerDomain where LocalSystem doesn't have rights to the domain

Can you reduce this line down to 80 chars - one of the tests is failing (https://ci.appveyor.com/project/PowerShell/xcomputermanagement/build/1.9.326.0#L385) because the README.MD lines need to be a maximum of 80 chars.

Just do something like this:

  * Resolved bug in Get-ComputerDomain where LocalSystem doesn't have rights
  to the domain

________________________________

DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 568 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhicDOVPYVg7yPkqu5:-KyhicDOVPYVg7yPkqu6:b2v8blg> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L568>):

        else
        {
            if((Get-WmiObject -Class Win32_ComputerSystem).PartOfDomain -eq $true)

WMI cmdlets is being phased out in favor of CIM cmdlets. So can you instead use:

Get-CimInstance -ClassName Win32_ComputerSystem

What you could also do to improve performance is to assign the result of the Get-CimInstance to a variable and use that. E.g.

$computerSystem = Get-CimInstance -ClassName Win32_ComputerSystem
if ($computerSystem.PartOfDomain -eq $true)
{
  $domainName = $computerSystem.Domain
}
else
{
  $domainName = $null
}

This will reduce the number of CIM lookups to 1 instead of 2. CIM (and WMI) lookups are quite slow.

Also, please add a space after if and before open bracket.

________________________________

Comments from Reviewable<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-KyhhCRQaSBVmIgLtZYE:bk39kn>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/PowerShell/xComputerManagement/pull/113#issuecomment-343706653>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APkgRvoXouCLrBJblxvivR_ecnIkQeFUks5s1kgHgaJpZM4QajBW>.

markgossa commented:
Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[README.md, line 216 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhmrqLAy509wcjHeh:-KzQBobVeCFcJUbcXyhZ:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L216)):*
<details><summary><i>Previously, markgossa (Mark Gossa) wrote…</i></summary><blockquote>

Hi Daniel,

That's great thanks. I started learning Pester yesterday and will give it a go.

The other changes are quite straight forward so should be okay.

I'll send you another pull request when I have this all done.

Thanks.

Mark


-------- Original message -------- From: Daniel Scott-Raynsford <notifications@github.com> Date: 12/11/2017 01:21 (GMT+00:00) To: PowerShell/xComputerManagement <xComputerManagement@noreply.github.com> Cc: Mark Gossa <mark@gossa.co.uk>, Mention <mention@noreply.github.com> Subject: Re: [PowerShell/xComputerManagement] xComputer: Issue 44 (#113)


Really great job @markgossa<https://github.com/markgossa> - for a first PR this is really great work! 😁

Couple of minor tweaks, nothing major at all.

There is one hard part to do though 😁 this new code you've added is really good but it has now actually made this function testable! So we need to add some unit tests for it - and these are usually the trickiest bit.

The tests will allow us to validate that the code works as expected (a.k.a. functional testing) as well in future that it continues to work (a.k.a. regression testing). We use a Pester (a PS Module) to execute the tests. Here is a great article to help you get started: http://duffney.io/GettingStartedWithPester .

So what you'll need to do is add some unit tests to this file here: https://github.com/PowerShell/xComputerManagement/blob/5a468b311115b9c6314e4576d4f0ad97ba0fe7a5/Tests/Unit/MSFT_xComputer.Tests.ps1#L928

You'd add a section something like this:

Context "$($script:DSCResourceName)\Get-ComputerDomain" {         It 'Should not throw when NetBios is not specified' {                 {                         Get-ComputerDomain -Verbose                 } | Should Not Throw         }

        It 'Should ....' {
                # Tests         }

        # More tests here...

        It 'Should not throw when NetBios is specified' {                 {                         Get-ComputerDomain -NetBios -Verbose                 } | Should Not Throw         } }

This will then cause the unit test coverage to increase. I'm more than happy to help you with all this as I know Pester can have a bit of a steep learning curve (but well worth learning). I apologize if you already know all this 😁

________________________________

Reviewed 2 of 2 files at r1. Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

________________________________

README.md, line 216 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhmrqLAy509wcjHeh:-KyhhmrrHcvnNdWGx0-s:b-qzsni0> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L216>):

* xOfflineDomainJoin:
  * Updated to meet HQRM guidelines.

Can you please remove this blank line - it'll trip up the automated release note generation up 😁

________________________________

README.md, line 217 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhgTneI983-sUcM8I:-KyhhgToFxkkCzfB6_mg:bivf2nk> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L217>):

  * Updated to meet HQRM guidelines.

* xComputer

Can you pop a colon on the end of this line for consistency.

________________________________

README.md, line 218 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhiZzPhSoBWbvILPW6:-KyhiZzPhSoBWbvILPW7:b-33f7m4> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L218>):

* xComputer
  * Resolved bug in Get-ComputerDomain where LocalSystem doesn't have rights to the domain

Can you reduce this line down to 80 chars - one of the tests is failing (https://ci.appveyor.com/project/PowerShell/xcomputermanagement/build/1.9.326.0#L385) because the README.MD lines need to be a maximum of 80 chars.

Just do something like this:

  * Resolved bug in Get-ComputerDomain where LocalSystem doesn't have rights   to the domain

________________________________

DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 568 at r1<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhicDOVPYVg7yPkqu5:-KyhicDOVPYVg7yPkqu6:b2v8blg> (raw file<https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L568>):

        else         {             if((Get-WmiObject -Class Win32_ComputerSystem).PartOfDomain -eq $true)

WMI cmdlets is being phased out in favor of CIM cmdlets. So can you instead use:

Get-CimInstance -ClassName Win32_ComputerSystem

What you could also do to improve performance is to assign the result of the Get-CimInstance to a variable and use that. E.g.

$computerSystem = Get-CimInstance -ClassName Win32_ComputerSystem if ($computerSystem.PartOfDomain -eq $true) {   $domainName = $computerSystem.Domain } else {   $domainName = $null }

This will reduce the number of CIM lookups to 1 instead of 2. CIM (and WMI) lookups are quite slow.

Also, please add a space after if and before open bracket.

________________________________

Comments from Reviewable<https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-KyhhCRQaSBVmIgLtZYE:bk39kn>

— You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub<https://github.com/PowerShell/xComputerManagement/pull/113#issuecomment-343706653>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APkgRvoXouCLrBJblxvivR_ecnIkQeFUks5s1kgHgaJpZM4QajBW>. 
</blockquote></details>

Done.

---

*[README.md, line 217 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhhgTneI983-sUcM8I:-KzQ85wjlUniBV9oF6Cv:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L217)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you pop a colon on the end of this line for consistency.
</blockquote></details>

Done.

---

*[README.md, line 218 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhiZzPhSoBWbvILPW6:-KzQ87Qc9NTd-JVhy4HI:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/README.md#L218)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you reduce this line down to 80 chars - one of the tests is failing (https://ci.appveyor.com/project/PowerShell/xcomputermanagement/build/1.9.326.0#L385) because the README.MD lines need to be a maximum of 80 chars.

Just do something like this:
```markdown
  * Resolved bug in Get-ComputerDomain where LocalSystem doesn't have rights
  to the domain
```
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 568 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-KyhicDOVPYVg7yPkqu5:-KzQBdfz7LVUiDzn7QYk:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/69829caaa47f32e0e602a9670dae6726a4abfb2e/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L568)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

WMI cmdlets is being phased out in favor of CIM cmdlets. So can you instead use:
```powershell
Get-CimInstance -ClassName Win32_ComputerSystem
```

What you could also do to improve performance is to assign the result of the `Get-CimInstance` to a variable and use that. E.g.
```powershell
$computerSystem = Get-CimInstance -ClassName Win32_ComputerSystem
if ($computerSystem.PartOfDomain -eq $true)
{
  $domainName = $computerSystem.Domain
}
else
{
  $domainName = $null
}
```

This will reduce the number of CIM lookups to 1 instead of 2. CIM (and WMI) lookups are quite slow.

Also, please add a space after if and before open bracket.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113)*
<!-- Sent from Reviewable.io -->

markgossa commented:
Still working on this and am nearly done.

---

Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-KzQCbhZRsdk5vpeYlI8:b-cwohr5)*
<!-- Sent from Reviewable.io -->

markgossa commented:
Hi @PlagueHO ,

I made the required changes to MSFT_xComputer.psm1 but there's a conflict in README.md. Should be simple to resolve. 

I'm still working on the new unit tests and will commit these soon.

Thanks.

Mark

markgossa commented:
@PlagueHO - All changes made as requested and committed along with new unit tests for Get-ComputerDomain.

Pester is good fun! I actually found another bug with that function so resolved that also. 

Let me know if there's anything else that needs to be done to have this pull request accepted.

PlagueHO commented:
@markgossa - really awesome job on the pester tests! Looking great. I just made some minor nitpicks (the usual things I miss as well :grin: ). So if you can get those changed I'll merge. Thanks again for all your work and patience! 

---

Reviewed 2 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 576 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TRXgfwaxtCF5JWecE:-L-TRXgfwaxtCF5JWecF:b-2ngw6s) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L576)):*
> ```PowerShell
>         else
>         {
>             $domainName = ""
> ```

Can you convert this to use single quotes (they are evaluated faster by PS)?

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 931 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TRsZKZkkZOukA71qv:-L-TRsZKZkkZOukA71qw:bs5ia7f) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L931)):*
> ```PowerShell
> 
>             Context "$($script:DSCResourceName)\Get-ComputerDomain" {
>                 It "returns domain netbios or DNS name if domain member" {
> ```

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).
2. Use capital 'r' in 'Returns'

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 932 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TS78j_JC1rXWlBZVy:-L-TS78j_JC1rXWlBZVz:bj34su0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L932)):*
> ```PowerShell
>             Context "$($script:DSCResourceName)\Get-ComputerDomain" {
>                 It "returns domain netbios or DNS name if domain member" {
>                     Mock -CommandName Get-CimInstance -ParameterFilter {$ClassName -eq "Win32_ComputerSystem"} -MockWith {
> ```

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 939 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSJ4AdSagKx-5VtSZ:-L-TSJ4AdSagKx-5VtS_:bj34su0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L939)):*
> ```PowerShell
>                     }
> 
>                     Mock -CommandName Get-Item -ParameterFilter {$Path -eq "Env:\USERDOMAIN"} -MockWith {
> ```

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 949 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSbB2pdBb6DuB9lS7:-L-TSbB34pjseM20wsyV:b-79gmbg) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L949)):*
> ```PowerShell
>                     }
> 
>                     Get-ComputerDomain @getComputerDomainParameters | Should Be "CONTOSO"
> ```

Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 955 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSbqXHkejSD7QG4cR:-L-TSbqXHkejSD7QG4cS:b-79gmbg) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L955)):*
> ```PowerShell
>                     }
> 
>                     Get-ComputerDomain @getComputerDomainParameters | Should Be "contoso.com"
> ```

Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 961 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSe0wM_McVtUmnEeu:-L-TSe0wM_McVtUmnEev:bs5ia7f) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L961)):*
> ```PowerShell
>                 }
> 
>                 It "returns nothing if in a workgroup" {
> ```

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).
2. Use capital 'r' in 'Returns'

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 962 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTAVKOdwZvPCtfkfq:-L-TTAVKOdwZvPCtfkfr:bj34su0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L962)):*
> ```PowerShell
> 
>                 It "returns nothing if in a workgroup" {
>                     Mock -CommandName Get-CimInstance -ParameterFilter {$ClassName -eq "Win32_ComputerSystem"} -MockWith {
> ```

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 969 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTHXOxAW8YWqEAL3d:-L-TTHXOxAW8YWqEAL3e:bj34su0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L969)):*
> ```PowerShell
>                     }
> 
>                     Mock -CommandName Get-Item -ParameterFilter {$Path -eq "Env:\USERDOMAIN"} -MockWith {
> ```

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 979 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSgE5NpOwdu9KUob0:-L-TSgE5NpOwdu9KUob1:b1hwnka) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L979)):*
> ```PowerShell
>                     }
> 
>                     Get-ComputerDomain @getComputerDomainParameters | Should Be ""
> ```

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 985 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TT6D_4NGmbZIr35vQ:-L-TT6D_4NGmbZIr35vR:b1hwnka) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L985)):*
> ```PowerShell
>                     }
> 
>                     Get-ComputerDomain @getComputerDomainParameters | Should Be ""
> ```

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 992 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTBxu42kcsUazoV6z:-L-TTBxu42kcsUazoV7-:bj34su0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L992)):*
> ```PowerShell
> 
>                 It "returns domain DNS name when netbios not specified" {
>                     Mock -CommandName Get-CimInstance -ParameterFilter {$ClassName -eq "Win32_ComputerSystem"} -MockWith {
> ```

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 999 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTDBM18C1NCyahxBb:-L-TTDBM18C1NCyahxBc:bj34su0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L999)):*
> ```PowerShell
>                     }
> 
>                     Mock -CommandName Get-Item -ParameterFilter {$Path -eq "Env:\USERDOMAIN"} -MockWith {
> ```

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-L-TTNk0pfaei0YDd_gj:b-mlxgnj)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Hi @markgossa - I changed the title of this PR to add the 'Fixes #44' to the end so that Github will automatically close the issue when this PR completes. Let me know when you're ready for me to continue the review and we'll get this one done! :grin:

markgossa commented:
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 576 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TRXgfwaxtCF5JWecE:-L0-R_I1Y9_jbMVjUAjL:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L576)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you convert this to use single quotes (they are evaluated faster by PS)?
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 931 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TRsZKZkkZOukA71qv:-L0-RjDkFAIeQn5kA40C:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L931)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).
2. Use capital 'r' in 'Returns'

</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 932 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TS78j_JC1rXWlBZVy:-L0-SlVQgPWw0_oEhRRh:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L932)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 939 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSJ4AdSagKx-5VtSZ:-L0-T0rsFtYyWnbmCcLC:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L939)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 949 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSbB2pdBb6DuB9lS7:-L0-T5xrzb3_YB7lQHMV:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L949)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 955 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSbqXHkejSD7QG4cR:-L0-TEZBNbt2QRhe2_c6:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L955)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 961 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSe0wM_McVtUmnEeu:-L0-TLAQOja6TNrRGqUJ:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L961)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).
2. Use capital 'r' in 'Returns'

</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 962 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTAVKOdwZvPCtfkfq:-L0-TgTYOjluB7cqHTq_:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L962)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 969 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTHXOxAW8YWqEAL3d:-L0-Tm51xo14nyvxUUNP:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L969)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 979 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TSgE5NpOwdu9KUob0:-L0-TqoKgUkLko6uSXXM:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L979)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).

</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 985 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TT6D_4NGmbZIr35vQ:-L0-TudWM2BWz5grqXal:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L985)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Change double quotes to use single quotes (faster when not containing variables).

</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 992 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTBxu42kcsUazoV6z:-L0-U-g13EbpsGJYrjyF:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L992)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 999 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L-TTDBM18C1NCyahxBb:-L0-U4-1RPLRRtEKcgXq:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/b80c7fab4e2929bacf69392405f1da909e2c2d0d/Tests/Unit/MSFT_xComputer.Tests.ps1#L999)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you:
1. Add spaces around { } to improve readability.
2. Change double quotes to use single quotes (faster when not containing variables).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113)*
<!-- Sent from Reviewable.io -->

markgossa commented:
Hi @PlagueHO - I've made the required changes and am ready for you to continue the review

PlagueHO commented:
Hi @markgossa - are you able to resolve the conflicts on this one and I'll complete the review? Thank you :smile:

markgossa commented:
Done.

---

Review status: 0 of 3 files reviewed at latest revision, 13 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-L0_7HxwFJgyWmhXx7er:b-896fix)*
<!-- Sent from Reviewable.io -->

markgossa commented:
Hi @PlagueHO - I've resolved the conflicts in README.md for you now and looks like it's passed the checks too. Thanks.

PlagueHO commented:
Hi @markgossa - Nearly there - just need to merge the changes from Dev branch and then move your change entry back into the Unreleased section and we'll be good to go. Thank you so much for your work on this and patience with me getting this through :grin:

---

Reviewed 2 of 3 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[README.md, line 238 at r6](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L1-PGly35HJs54SgOZp:-L1-PGlzhPdLAUzVNPpX:b7ire7o) ([raw file](https://github.com/powershell/xcomputermanagement/blob/dc158b62a8a43db5ec93a445e4ee258196b272ed/README.md#L238)):*
> ```Markdown
>   * Correct `Assert-VerifiableMocks` to `Assert-VerifiableMock`
> 
> * xComputer:
> ```

Unfortunately the resource kit was released and so the change log has been changed a bit :cry: which made your change show up in the wrong place.

So can you merge the changes from the upstream Dev branch and then move this up into the Unreleased section? After that we are good to go and I'll merge.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-L1-PdWZS2-jJnpV1Rn3:b-o1nde8)*
<!-- Sent from Reviewable.io -->

markgossa commented:
Done.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[README.md, line 238 at r6](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-L1-PGly35HJs54SgOZp:-L19RKVBaZkYe-R4JaJe:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/dc158b62a8a43db5ec93a445e4ee258196b272ed/README.md#L238)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Unfortunately the resource kit was released and so the change log has been changed a bit :cry: which made your change show up in the wrong place.

So can you merge the changes from the upstream Dev branch and then move this up into the Unreleased section? After that we are good to go and I'll merge.

</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-L19RGEeJyREt-tNPBGj:b-896fix)*
<!-- Sent from Reviewable.io -->

markgossa commented:
Hi @PlagueHO - I've made the changes you asked for and hopefully it's good to go with the merge. If there's anything else you need changed, just let me know. Thanks.

PlagueHO commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Nice job @markgossa!

---

Reviewed 2 of 3 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/113#-:-L1AdbqMDVDZiHscEh30:bxfjbcw)*
<!-- Sent from Reviewable.io -->

markgossa commented:
@PlagueHO - thanks and Merry Christmas! Maybe I'll meet you on another issue.

PlagueHO commented:
You too @markgossa - I look forward to your next contribution! 😃 Keep up the really great stuff!
